src/main/resources/assets/computercraft/lua/rom/apis/term.lua

```diff
@@ -12,17 +12,15 @@ local term = {}
 
 term.redirect = function( target )
 	if target == nil or type( target ) ~= "table" then
-		error( "Invalid redirect target", 2 )
+		error( "bad argument #1 (expected table, got " .. type( target ) .. ")", 2 )
 	end
     if target == term then
         error( "term is not a recommended redirect target, try term.current() instead", 2 )
     end
 	for k,v in pairs( native ) do
 		if type( k ) == "string" and type( v ) == "function" then
 			if type( target[k] ) ~= "function" then
-				target[k] = function()
-					error( "Redirect object is missing method "..k..".", 2 )
-				end
+				error( "Redirect object is missing method "..k..".", 2 )
```

Comment:
Yeah, don't redirect CraftOS to a terminal object unless you know it's suitable for the purpose.

All you're doing here is shifting the crash from a "might happen" to a "will happen" situation. If the terminal object isn't suitable, then *that's* the problem that needs to be addressed.